ocp_resources/ip_address_pool.py

```diff
@@ -0,0 +1,52 @@
+# API reference: https://metallb.universe.tf/apis/#metallb.io/v1beta1.IPAddressPool
+
+from ocp_resources.resource import NamespacedResource
+
+
+class IPAddressPool(NamespacedResource):
+    """
+    IPAddressPool object.
+    """
+
+    api_group = NamespacedResource.ApiGroup.METALLB_IO
+
+    def __init__(
+        self,
+        name=None,
+        namespace=None,
+        client=None,
+        addresses=None,
+        auto_assign=True,
+        avoid_buggy_ips=False,
+        **kwargs,
+    ):
+        """
+        Args:
+            name (str): Name of the MetalLB or it's CR's.
+            namespace (str): Namespace of the MetalLB
+            client: (DynamicClient): DynamicClient to use.
+            addresses (list): List of IP addresses.
+            auto_assign (bool): IP address assignment. (default: True)
+            avoid_buggy_ips (bool): Avoid buggy IP address (default: False)
+        """
+        super().__init__(
+            name=name,
+            namespace=namespace,
+            client=client,
+            **kwargs,
+        )
+        self.addresses = addresses
```

Comment:
You need to check if addresses is not None and raise if it is None

Reply:
Okay I will update it.

Reply:
Updated it.